### PR TITLE
Added GZIP support for static files

### PR DESCRIPTION
**Pull request by perwendel**
Also added test cases for all gzip support and Gzip Example.

### CODE REVIEW

File: src/main/java/spark/staticfiles/StaticFiles.java [hunk 1]
```diff
@@ -53,12 +53,15 @@
...
      */
     public static boolean consume(HttpServletRequest httpRequest,
-                                  ServletResponse servletResponse) throws IOException {
+                                  HttpServletResponse httpResponse) throws IOException {
         if (staticResourceHandlers != null) {
             for (AbstractResourceHandler staticResourceHandler : staticResourceHandlers) {
                 AbstractFileResolvingResource resource = staticResourceHandler.getResource(httpRequest);
                 if (resource != null && resource.isReadable()) {
-                    IOUtils.copy(resource.getInputStream(), servletResponse.getOutputStream());
+                    OutputStream wrappedOutputStream = GzipUtils.checkAndWrap(httpRequest, httpResponse, false);
```
Comment (tipsy) [inline src/main/java/spark/staticfiles/StaticFiles.java]: Name?

File: src/main/java/spark/utils/GzipUtils.java [hunk 2]
```diff
@@ -49,22 +49,27 @@ private GzipUtils() {
...
 
         // GZIP Support handled here. First we must ensure that we want to use gzip, and that the client supports gzip
         boolean acceptsGzip = Collections.list(httpRequest.getHeaders(ACCEPT_ENCODING)).stream().anyMatch(STRING_MATCH);
         boolean wantGzip = httpResponse.getHeaders(CONTENT_ENCODING).contains(GZIP);
 
-        if (acceptsGzip && wantGzip) {
-            outputStream = new GZIPOutputStream(outputStream, true);
+        if (acceptsGzip) {
+            if (wantGzip || !requireWantsHeader) {
+                outputStream = new GZIPOutputStream(outputStream, true);
```
Comment (tipsy) [inline src/main/java/spark/utils/GzipUtils.java]: Name?

File: src/main/java/spark/webserver/MatcherFilter.java [hunk 3]
```diff
@@ -260,12 +259,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
                 }
 
                 // Check if gzip is wanted/accepted and in that case handle that
-                OutputStream outputStream = GzipUtils.checkAndWrap(httpRequest, httpResponse);
+                OutputStream outputStream = GzipUtils.checkAndWrap(httpRequest, httpResponse, true);
```
Comment (tipsy) [inline src/main/java/spark/webserver/MatcherFilter.java]: Name!

File: src/main/java/spark/webserver/MatcherFilter.java [hunk 4]
```diff
@@ -260,12 +259,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
```
Comment (tipsy) [inline src/main/java/spark/webserver/MatcherFilter.java]: Name!

File: src/main/java/spark/utils/GzipUtils.java [hunk 5]
```diff
@@ -49,22 +49,27 @@ private GzipUtils() {
...
+                                                                        IOException {
         OutputStream outputStream = httpResponse.getOutputStream();
 
         // GZIP Support handled here. First we must ensure that we want to use gzip, and that the client supports gzip
         boolean acceptsGzip = Collections.list(httpRequest.getHeaders(ACCEPT_ENCODING)).stream().anyMatch(STRING_MATCH);
         boolean wantGzip = httpResponse.getHeaders(CONTENT_ENCODING).contains(GZIP);
 
-        if (acceptsGzip && wantGzip) {
-            outputStream = new GZIPOutputStream(outputStream, true);
+        if (acceptsGzip) {
```
Comment (tipsy) [inline src/main/java/spark/utils/GzipUtils.java]: Flip?

Comment (tipsy) [inline src/main/java/spark/utils/GzipUtils.java]: Combine (or flip) ?

Comment (perwendel) [inline src/main/java/spark/utils/GzipUtils.java]: flipped!

**Pull request merged**